### PR TITLE
Customloader unit tests fixes

### PR DESCRIPTION
**Pull request by ravikumarbhattiprolu**
All,
   Cleaned up reused code in custom loader unit tests and moved the needed code into a header. The same header is used by existing tests and customloader unit tests. Please review the changes.
regards,
Ravi

### CODE REVIEW

File: src/test/prediction_service_test.hpp [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+#include "../executinstreamidguard.hpp"
+#include "../modelinstance.hpp"
+#include "../prediction_service_utils.hpp"
+#include "test_utils.hpp"
+
+using testing::Each;
+using testing::Eq;
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wnarrowing"
```
Comment (atobiszei) [inline src/test/prediction_service_test.hpp:43]: Why you need those? I think we needed those only for includes from TensorflowServing.

File: src/test/custom_loader_test.cpp [hunk 2]
```diff
@@ -211,24 +178,9 @@ const char* custom_loader_config_model_multiple = R"({
...
-public:
-    std::shared_ptr<ovms::Model> modelFactory(const std::string& name) override {
-        return modelMock;
-    }
-};
-
-class TestCustomLoader : public ::testing::Test {
+class TestCustomLoader : public TestPredict {
 public:
     void SetUp() {
```
Comment (atobiszei) [inline src/test/custom_loader_test.cpp:185]: You made this child class of TestPredict but you are not calling base class SetUp method which I think is not intended:

you most probably should call:
TestPredict::Setup().

cleaning up temporary destination most probably should be done in TearDown method?(at the end of test)

Comment (ravikumarbhattiprolu) [reply src/test/custom_loader_test.cpp:185]: The Setup in original calls TestPredict::Setup() sets up model instance and that is not used in customloader tests. these tests start from file. The Setup in derived class does the needed for customloader tests. So, we did not call base class setup.

The teardown actually deletes the files and does cleanup. We have put the other cleanup call in setup just incase preious test execution returned abnormally and some files remained.

File: src/test/prediction_service_test.hpp (hunk 1 above)
Comment (rasapala) [reply src/test/prediction_service_test.hpp:43]: Some tests also needed exceptions in warnings.

**Comment by dtrawins** [comment]
out of date

**Pull request closed without merging**